include/z64scene.h

```diff
@@ -417,4 +419,132 @@ typedef enum {
     /* 0x6E */ SCENE_ID_MAX
 } SceneID;
 
+// Scene commands
+
+typedef enum {
+    /* 0x00 */ SCENE_CMD_ID_SPAWN_LIST,
+    /* 0x01 */ SCENE_CMD_ID_ACTOR_LIST,
+    /* 0x02 */ SCENE_CMD_ID_ACTOR_CUTSCENE_CAM_LIST,
+    /* 0x03 */ SCENE_CMD_ID_COL_HEADER,
+    /* 0x04 */ SCENE_CMD_ID_ROOM_LIST,
+    /* 0x05 */ SCENE_CMD_ID_WIND_SETTINGS,
+    /* 0x06 */ SCENE_CMD_ID_ENTRANCE_LIST,
+    /* 0x07 */ SCENE_CMD_ID_SPECIAL_FILES,
+    /* 0x08 */ SCENE_CMD_ID_ROOM_BEHAVIOR,
+    /* 0x09 */ SCENE_CMD_ID_UNUSED_09,
+    /* 0x0A */ SCENE_CMD_ID_MESH,
+    /* 0x0B */ SCENE_CMD_ID_OBJECT_LIST,
+    /* 0x0C */ SCENE_CMD_ID_LIGHT_LIST,
+    /* 0x0D */ SCENE_CMD_ID_PATH_LIST,
+    /* 0x0E */ SCENE_CMD_ID_TRANSI_ACTOR_LIST,
+    /* 0x0F */ SCENE_CMD_ID_ENV_LIGHT_SETTINGS,
+    /* 0x10 */ SCENE_CMD_ID_TIME_SETTINGS,
+    /* 0x11 */ SCENE_CMD_ID_SKYBOX_SETTINGS,
+    /* 0x12 */ SCENE_CMD_ID_SKYBOX_DISABLES,
+    /* 0x13 */ SCENE_CMD_ID_EXIT_LIST,
+    /* 0x14 */ SCENE_CMD_ID_END,
+    /* 0x15 */ SCENE_CMD_ID_SOUND_SETTINGS,
+    /* 0x16 */ SCENE_CMD_ID_ECHO_SETTINGS,
+    /* 0x17 */ SCENE_CMD_ID_CUTSCENE_LIST,
+    /* 0x18 */ SCENE_CMD_ID_ALTERNATE_HEADER_LIST,
+    /* 0x19 */ SCENE_CMD_ID_MISC_SETTINGS,
+    /* 0x1A */ SCENE_CMD_ID_TEXTURE_ANIM_LIST,
+    /* 0x1B */ SCENE_CMD_ID_ACTOR_CUTSCENE_LIST,
+    /* 0x1C */ SCENE_CMD_ID_MINIMAP_INFO,
+    /* 0x1D */ SCENE_CMD_ID_UNUSED_1D,
+    /* 0x1E */ SCENE_CMD_ID_MINIMAP_COMPASS_ICON_INFO
+} SceneCommandTypeID;
+
+#define SCENE_CMD_SPAWN_LIST(numSpawns, spawnList) \
+    { SCENE_CMD_ID_SPAWN_LIST, numSpawns, CMD_PTR(spawnList) }
+
+#define SCENE_CMD_ACTOR_LIST(numActors, actorList) \
+    { SCENE_CMD_ID_ACTOR_LIST, numActors, CMD_PTR(actorList) }
+
+#define SCENE_CMD_CAM_LIST(numCams, camList) \
+    { SCENE_CMD_ID_ACTOR_CUTSCENE_CAM_LIST, numCams, CMD_PTR(camList) }
+
+#define SCENE_CMD_COL_HEADER(colHeader) \
+    { SCENE_CMD_ID_COL_HEADER, 0, CMD_PTR(colHeader) }
+
+#define SCENE_CMD_ROOM_LIST(numRooms, roomList) \
+    { SCENE_CMD_ID_ROOM_LIST, numRooms, CMD_PTR(roomList) }
+
+#define SCENE_CMD_WIND_SETTINGS(xDir, yDir, zDir, strength) \
+    { SCENE_CMD_ID_WIND_SETTINGS, 0, CMD_BBBB(xDir, yDir, zDir, strength) }
+
+#define SCENE_CMD_ENTRANCE_LIST(entranceList) \
+    { SCENE_CMD_ID_ENTRANCE_LIST, 0, CMD_PTR(entranceList) }
+
+#define SCENE_CMD_SPECIAL_FILES(elfMessageFile, keepObjectId) \
+    { SCENE_CMD_ID_SPECIAL_FILES, elfMessageFile, CMD_W(keepObjectId) }
+
+#define SCENE_CMD_ROOM_BEHAVIOR(curRoomUnk3, curRoomUnk2, showInvisActors, msgCtxUnkE40C) \
+    { SCENE_CMD_ID_ROOM_BEHAVIOR, curRoomUnk3, \
+        curRoomUnk2 | _SHIFTL(showInvisActors, 8, 1) | _SHIFTL(msgCtxUnkE40C, 10, 1) }
```

Comment:
About this macro, since `showInvisActors` and `msgCtxUnkE40C` are 1-bit flags, would make more sense if ZAPD outputted `true` and `false` for those?

For example:
```c
static SCmdBase ganontika_room_16Set0000[8] = {
	SCENE_CMD_ECHO_SETTINGS(6),
	SCENE_CMD_ROOM_BEHAVIOR(0x01, 0x00, true, false),
	SCENE_CMD_SKYBOX_DISABLES(0, 1),
	SCENE_CMD_TIME_SETTINGS(255, 255, 0),
	SCENE_CMD_MESH(&ganontika_room_16PolygonType2_0000B0),
	SCENE_CMD_OBJECT_LIST(5, ganontika_room_16ObjectList_000040),
	SCENE_CMD_ACTOR_LIST(6, ganontika_room_16ActorList_00004C),
	SCENE_CMD_END(),
}; 
```
instead of 
```c
static SCmdBase ganontika_room_16Set0000[8] = {
	SCENE_CMD_ECHO_SETTINGS(6),
	SCENE_CMD_ROOM_BEHAVIOR(0x01, 0x00, 1, 0),
	SCENE_CMD_SKYBOX_DISABLES(0, 1),
	SCENE_CMD_TIME_SETTINGS(255, 255, 0),
	SCENE_CMD_MESH(&ganontika_room_16PolygonType2_0000B0),
	SCENE_CMD_OBJECT_LIST(5, ganontika_room_16ObjectList_000040),
	SCENE_CMD_ACTOR_LIST(6, ganontika_room_16ActorList_00004C),
	SCENE_CMD_END(),
}; 
```

Reply:
for showinvis actors id say yes. not sure about the message context one though, cause im not sure what that variable is for

Reply:
the same can apply to skybox disables

Reply:
according to tharos message decomp, `msgCtxUnkE40C` is `disableWarpSongs`. can rename the arg and use true/false